src/api/spec/controllers/staging/staging_projects_controller_spec.rb

```diff
@@ -139,4 +139,44 @@
                                                                                                            user.id)
     end
   end
+
+  describe 'POST #create' do
+    let(:staging_workflow_project) { staging_workflow.project.name }
+    let!(:other_project) { create(:project, name: "#{project}:other_project") }
+
+    before do
+      login(user)
+      staging_workflow
+    end
+
+    subject { post :create, params: { staging_workflow_project: staging_workflow_project, format: :xml }, body: body }
+
+    context 'succeeds' do
+      let(:body) do
+        <<~XML
+          <workflow>
+            <staging_project>#{project}:Staging:C</staging_project>
+            <staging_project>#{project}:other_project</staging_project>
+          </workflow>
+        XML
+      end
+
+      it { expect(subject).to have_http_status(:success) }
+      it { expect { subject }.to change(Project, :count).by(1) }
+    end
+
+    context 'fails: project already assigned to a staging workflow' do
+      let(:body) do
+        <<~XML
+          <workflow>
+            <staging_project>#{project}:Staging:A</staging_project>
+            <staging_project>#{project}:Staging:E</staging_project>
+          </workflow>
+        XML
+      end
+
+      it { expect(subject).to have_http_status(:bad_request) }
+      it { expect { subject }.not_to change(Project, :count) }
+    end
+  end
```

Comment:
What happens if I don't send a body?

Reply:
This would have caused a failure. It's fixed now.

Reply:
the ```StagingProjectCreator``` won't be valid, and therefore, it will raise a ```bad_request```